### PR TITLE
[IMP] core: improve server-side Form

### PR DESCRIPTION
**Pull request by rco-odoo**
Improve the implementation of the server-side `Form` to make it more readable.
Make it also more consistent with the actual form view implementation in the web client:
* the call to `onchange()` should use the context of the field that triggers it;
* the views are retrieved with method `get_views()`.

### CODE REVIEW

File: odoo/tests/__init__.py [hunk 1]
```diff
@@ -5,6 +5,6 @@
 See https://github.com/python/cpython/tree/3.10/Lib/unittest for reference files.
 """
 
-from . import common
+from . import common, form
```
[the remark this answers is not in the document]
Comment (rco-odoo) [reply odoo/tests/__init__.py:8]: Maybe we should recommend using `from odoo.tests import Form` instead :+1:

File: odoo/tests/form.py [hunk 2]
```diff
@@ -793,6 +804,17 @@ def update(self, *args, **kw):
             self._changed.update(args[0]._changed)
 
 
+class Dotter:
```
[the remark this answers is not in the document]
Comment (ryv-odoo) [reply odoo/tests/form.py]: Sadly, as we discussed with @xmo-odoo , the `DotDict` is probably unsafe at this place: 

The `Form` is used in the business code (at least one time: https://github.com/odoo/enterprise/blob/15ddcb297fd31740bdb3ac275d592a66f116774f/account_taxcloud/models/account_invoice.py#L119 ), the evaluation of this safe_eval is done on data that can be modified by the admin user, and the DotDict is unsafe.

**Review by ryv-odoo — APPROVED** [review]
Expect for the DotDict (see the other comment), it looks good to me.

**Comment by rco-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**